tests/unittests/test_backoff_error_handling.py

```diff
@@ -0,0 +1,1782 @@
+        Test that tap retry for 60 seconds on the http timeout error.
+        '''
+        with open('tests/base.py') as f:
+            mock_get_account.side_effect = HTTPError('url', 408, 'Request Timeout', {}, f)
+        before_time = datetime.datetime.now()
+        try:
+            tap_bing_ads.sync_accounts_stream(['i1'], {})
+        except HTTPError:
+            pass
+        after_time = datetime.datetime.now()
+        time_difference = (after_time - before_time).total_seconds()
+        # verify the code backed off for 60 seconds
+        # time_difference should be greater or equall 60 as some time elapsed while calculating `after_time`
+        self.assertGreaterEqual(time_difference, 60)
+        
+    @mock.patch("tap_bing_ads.create_sdk_client", return_value = '')
+    @mock.patch("tap_bing_ads.CustomServiceClient")
+    def test_internal_server_error_get_account(self, mock_get_account, mock_create_sdk_client,
+                                                    mock_get_selected_fields, mock_get_core_schema,
+                                                    mock_write_schema, mock_get_bookmark,
+                                                    mock_sobject_to_dict, mock_write_state,
+                                                    mock_write_bookmark, mock_metrics, mock_write_records,
+                                                    mock_filter_selected_fields_many):
+        '''
+        Test that tap retry for 60 seconds on the 500 internal server error.
+        '''
+        with open('tests/base.py') as f:
+            mock_get_account.side_effect = HTTPError('url', 500, 'Internal Server Error', {}, f)
+        before_time = datetime.datetime.now()
+        try:
+            tap_bing_ads.sync_accounts_stream(['i1'], {})
+        except HTTPError:
+            pass
+        after_time = datetime.datetime.now()
+        time_difference = (after_time - before_time).total_seconds()
+        # verify the code backed off for 60 seconds
+        # time_difference should be greater or equall 60 as some time elapsed while calculating `after_time`
+        self.assertGreaterEqual(time_difference, 60)
+        
+    @mock.patch("tap_bing_ads.create_sdk_client", return_value = '')
+    @mock.patch("tap_bing_ads.CustomServiceClient")
+    def test_transport_error_get_account(self, mock_get_account, mock_create_sdk_client,
+                                                mock_get_selected_fields, mock_get_core_schema,
+                                                mock_write_schema, mock_get_bookmark,
+                                                mock_sobject_to_dict, mock_write_state, 
+                                                mock_write_bookmark, mock_metrics, mock_write_records, 
+                                                mock_filter_selected_fields_many):
+        '''
+        Test that tap retry for 60 seconds on the Transport error.
+        '''
+        with open('tests/base.py') as f:
```

Comment:
This file handler is unused in this test and also some other tests too. Please remove those.

Reply:
Updated